### PR TITLE
Update arrow to include pandas serialization

### PR DESCRIPTION
**Pull request by pcmoritz**

### CODE REVIEW

File: python/ray/worker.py [hunk 1]
```diff
@@ -1080,6 +1080,48 @@ def default_dict_custom_deserializer(obj):
...
         custom_deserializer=default_dict_custom_deserializer)
 
+    def _serialize_pandas_series(s):
+        import pandas as pd
+        # TODO: serializing Series without extra copy
+        serialized = serialize_pandas(pd.DataFrame({s.name: s}))
+        return {
+            'type': 'Series',
+            'data': serialized.to_pybytes()
+         }
```
Comment (robertnishihara) [inline python/ray/worker.py]: indentation is off here

File: python/ray/worker.py [hunk 2]
```diff
@@ -1080,6 +1080,48 @@ def default_dict_custom_deserializer(obj):
...
+
+
```
Comment (robertnishihara) [inline python/ray/worker.py]: too much white space between these functions, should be one line, run `flake8 worker.py`.

**Comment by robertnishihara** [comment]
We could also just use the pyarrow default context. And we could add our defaultdict and OrderedDict custom serializers to the default context. What do you think about that?

**Comment by pcmoritz** [comment]
Yeah, I'm planning on doing that and putting everything into the default context in arrow; this will require another PR to arrow.

**Review by robertnishihara — APPROVED** [review]

**Pull request merged**